File: September 2022/Day 14/JAVA Code/Power Of 2 and Subsequences.java
//{ Driver Code Starts
//Initial Template for Java
import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while(t-- > 0)
        {
            int N = Integer.parseInt(read.readLine());
            ArrayList<Long> A = new ArrayList<Long>();
            String in[] = read.readLine().trim().split(" ");
            for(var i : in){
                Long x = Long.parseLong(i);
                A.add(x);
            }

            Solution ob = new Solution();
            System.out.println(ob.numberOfSubsequences(N,A));
        }
    }
}
// } Driver Code Ends


//User function Template for Java
class Solution{
    public static final long mod = 1000000007;
    static Long numberOfSubsequences(int N, ArrayList<Long> A){
        // code here
        long count =0;
        for(long n : A) {
            if(number_of_set_bit(n)==1) {
                count++;
            }
        }
        return subtract(power(2,count),1);
    }
    
    public static long power(long a,long b) {
        if(b==0) return 1;
        
        long p = power(a,b/2)%mod;
        
        if(b%2==1) {
            // p = p*p*a
            p = multiply(p,multiply(p,a));
        } else {
            // p = p*p
            p = multiply(p,p);
        }
        
        return p;
    }
    
    public static int number_of_set_bit(long n) {
        int ct=0;
        while(n>0) {
            long rsbm = n&(-n);
            ct++;
            n-=rsbm;
        }
        return ct;
    }
    
    public static long multiply(long a,long b) {
        return ((a%mod)*(b%mod))%mod;
    }
    
    public static lon

File: September 2022/Day 14/C++ Code/Power Of 2 and Subsequences.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
class Solution{   
public:
    long long int mod = 1000000007;
    long long numberOfSubsequences(int N, long long A[]){
        // code here
        long long count =0;
        for(int i=0;i<N;i++) {
            if(number_of_set_bit(A[i])==1) {
                count++;
            }
        }
        // 2^count -1;
        return subtract(power(2,count),1);
    }
    long long power(long long int a,long long int b) {
        if(b==0) return 1;
        
        long long int p = power(a,b/2)%mod;
        
        if(b%2==1) {
            // p = p*p*a
            p = multiply(p,multiply(p,a));
        } else {
            // p = p*p
            p = multiply(p,p);
        }
        
        return p;
    }
    
    int number_of_set_bit(long long int n) {
        int ct=0;
        while(n>0) {
            long long rsbm = n&(-n);
            ct++;
            n-=rsbm;
        }
        return ct;
    }
    
    long long multiply(long long int a,long long int b) {
        return ((a%mod)*(b%mod))%mod;
    }
    
    long long subtract(long long int a,long long int b) {
        // from a to b
        return ((a%mod)-(b%mod)+mod)%mod;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N;
        cin >> N;
        long long A[N];
        for(int i=0;i<N;++i){
            cin >> A[i];
        }
        Solution ob;
        cout << ob.numberOfSubsequences(N,A) << endl;
    }
    return 0; 
} 
// } Driver Code Ends
